libs/Microsoft.MixedReality.WebRTC.Native/test/simple_interop.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "simple_interop.h"
+
+void SimpleInterop::Register(mrsPeerConnectionHandle pc) noexcept {
+  mrsPeerConnectionInteropCallbacks interop{};
+  interop.remote_audio_track_create_object = &RemoteAudioTrackCreateStatic;
+  interop.remote_video_track_create_object = &RemoteVideoTrackCreateStatic;
+  interop.data_channel_create_object = &DataChannelCreateStatic;
+  ASSERT_EQ(Result::kSuccess,
+            mrsPeerConnectionRegisterInteropCallbacks(pc, &interop));
+}
+
+void SimpleInterop::Unregister(mrsPeerConnectionHandle pc) noexcept {
+  mrsPeerConnectionInteropCallbacks interop{};
+  ASSERT_EQ(Result::kSuccess,
+            mrsPeerConnectionRegisterInteropCallbacks(pc, &interop));
+}
+
+void* SimpleInterop::CreateObject(ObjectType type) noexcept {
+  const uint32_t id = free_id_.fetch_add(1);
+  auto handle = std::make_unique<Handle>(Handle{this, type, id});
+  void* interop_handle = handle.get();
+  try {
+    auto lock = std::scoped_lock{objects_map_mutex_};
+    objects_map_.emplace(interop_handle, std::move(handle));
+  } catch (...) {
+  }
```

Comment:
nit: why the try-catch?

Reply:
Generally when working with the interop layer all functions are `noexcept` to prevent passing an exception through the interop layer. Here `std::unordered_map::emplace()` is not declared `noexcept` and might throw, so I'm catching that. Now that being said we should probably return `nullptr` as a failure in the `catch`, and `std::make_unique` can also throw so should be inside the `try { }` block

Reply:
Also - as said before - we shouldn't discard exceptions silently, at the very least we should print a generic error message.

Reply:
Note that this is a test utility, only available inside the test executable, and we cannot log with `RTC_LOG` here (not exported). I will ensure the test fails with `ASSERT_TRUE(false)` though.

Reply:
Fixed.